File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        l = 0,
        r = len(nums) - 1
        memo = {}
        for i, num in enumerate(nums):
            if num in memo:
                if abs(memo[num] - i) <= k:
                    return True
                else:
                    memo[num] = i
            else:
                memo[num] = i
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int, int> hash;
        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            int diff = target - num;
            if (hash.count(num) == 1) {
                return {hash[num], i};
            }
            hash[diff] = i;
        }
        return {};
    }
};
